packages/suite/src/middlewares/wallet/coinjoinMiddleware.ts

```diff
@@ -15,6 +17,17 @@ export const coinjoinMiddleware =
             CoinjoinBackendService.getInstances().forEach(b => b.cancel());
         }
 
+        // do not close success and critical phase modals when they are open, similar to discovery middleware
+        const { modal } = api.getState();
+        const allowedModals = ['coinjoin-success', 'critical-coinjoin-phase'];
+
+        if (
```

Comment:
```suggestion
       if (
            action.type === UI.CLOSE_UI_WINDOW &&
            'payload' in modal &&
            allowedModals.includes(modal.payload?.type)
        ) {
            return action;
        }
```

instead of typeforcing

Reply:
8492c6f13b70a1c12739bbf78fd685db6024bcdd